ex2/ex2.ts

```diff
@@ -0,0 +1,80 @@
+interface Pessoa {
+  id: number;
+  name: string;
+  bio: string;
+}
+
+const lista: Pessoa[] = [
+  { "id": 1, "name": "Ada Lovelace", "bio": "Ada Lovelace, foi uma matemática e escritora inglesa reconhecida por ter escrito o primeiro algoritmo para ser processado por uma máquina" },
+  { "id": 2, "name": "Alan Turing", "bio": "Alan Turing foi um matemático, cientista da computação, lógico, criptoanalista, filósofo e biólogo teórico britânico, ele é amplamente considerado o pai da ciência da computação teórica e da inteligência artificial" },
+  { "id": 3, "name": "Nikola Tesla", "bio": "Nikola Tesla foi um inventor, engenheiro eletrotécnico e engenheiro mecânico sérvio, mais conhecido por suas contribuições ao projeto do moderno sistema de fornecimento de eletricidade em corrente alternada." },
+  { "id": 4, "name": "Nicolau Copérnico", "bio": "Nicolau Copérnico foi um astrônomo e matemático polonês que desenvolveu a teoria heliocêntrica do Sistema Solar." }
+];
+
+/* 
+ * Paradigma funcional
+ */
+
+const getBio = (id: number, lista: Pessoa[]) => lista.find(p => p.id === id)?.bio;
+
+const getName = (id: number, lista: Pessoa[]) => lista.find(p => p.id === id)?.name;
+
+const removePessoa = (id: number, lista: Pessoa[]) => lista.filter(p => p.id !== id);
+
+const alteraPessoa = (id: number, lista: Pessoa[], name: string, bio: string) => lista.map(p => (p.id === id) ? { "id": id, "name": name, "bio": bio } : p);
+
+const imprimeLista = (lista: Pessoa[]) => lista.forEach(p => console.log(`${p.id}. ${p.name}`));
+
+console.log('1. Demonstracao funcional');
+console.log('-------------------------');
+console.log(`a) ${getBio(2, lista)}`);
+console.log(`b) ${getName(2, lista)}`);
+
+let aux = removePessoa(4, lista);
+console.log(`\nc) Pessoa 4 removida. Lista atualizada:`);
+imprimeLista(aux);
+
+aux = alteraPessoa(3, aux, 'Brendan Eich', 'Brendan Eich é um programador de computadores e executivo de tecnologia americano. Ele criou a linguagem de programação JavaScript e foi cofundador do projeto Mozilla, da Mozilla Foundation e da Mozilla Corporation');
+console.log(`\nd) Pessoa 3 alterada. Lista atualizada:`);
+imprimeLista(aux);
+
+/* 
+ * Paradigma imperativo
+ */
+
+function getBioImp(id: number): string {
```

Comment:
Com exceção de comentários, mensagens e erros, onde podemos manter conforme a linguagem que for mais viável ao contexto, é legal sempre unificar o idioma do nosso código. Escrevendo os métodos, variáveis, tipos, etc. todos em único idioma para facilitar o entendimento e manter a padronização.